source/extensions/filters/network/redis_proxy/proxy_filter.cc

```diff
@@ -96,6 +99,25 @@ void ProxyFilter::onAuth(PendingRequest& request, const std::string& password) {
   request.onResponse(std::move(response));
 }
 
+void ProxyFilter::onAuth(PendingRequest& request, const std::string& username,
+                         const std::string& password) {
+  Common::Redis::RespValuePtr response{new Common::Redis::RespValue()};
+  if (config_->downstream_auth_username_.empty() && config_->downstream_auth_password_.empty()) {
```

Comment:
I think we need to handle the case where password is config but username is empty.  In this case the "AUTH default <password>" should return OK.